### PR TITLE
updates for issue#68

### LINKED ISSUE #68: Consolidate and merge use cases, revise h2n sections
The following aspects are considered here:
a) use cases in section 3 need to be consolidated (to address the tsvwg comments)
    Media streaming (3.1) and Interactive media (3.2) can be combined. 
    Accommodate-Delay Traffic (3.3) and User Preferences (3.4) combined --> this can represent a traffic management option for flow control (i.e., the user prefers to prioritize flow 1 (UDP 4-tuple) over flow 2 belonging to the user.
    Remove 3.5 and 3.6 
b) Add the "flow control" description above to the Traffic Shaping/management in section 1.
c) Revise h2n section 5.1 to match the revision for flow preference (in b) and/or s2n metadata.
d) clarify "intra-flow". With the revisions in s2n (5.2), only priority of packet is visible and intra-flow (audio, video, etc) is not.
    On the other hand, a user may have multiple flows that are shaped and have preferences on that (see b above)

### PR DESCRIPTION
**Pull request by John-Kaippallimalil**

### CODE REVIEW

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 1]
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
...
 In networks where link conditions (between Client and Router) vary significantly at timescales well below the RTT, this results in unused (wasted) bandwidth at short timescales.
 There is some research {{5G-Octopus}} to indicate that media applications can obtain better QoE when sending at a higher rate (less conservative than current CCA) and the media application is willing to tolerate some packet loss or delay of low priority packets.
 Packet priority and tolerance to delay of packets in such a case would be provided on-path in a side channel associated to the downstream packet (B – in figure).
 The requirements for this server-to-network (S2N) metadata are described in {{server-network}}.
+Network shapers observe flows and apply policies to maximize performance but is not aware if there is a preference for one flow (UDP 4-tuple) over another flow belonging to the same user and network attachment (e.g., a subscriber connection, a 3GPP PDU session. See {{net-attach}} for more details).
+The client can provide the (access) Router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in figure).
```
Comment (sridharangirish91) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: Are we delving into inter-flow priorities, outside of the same UDP 4-tuple? I thought we mindfully wanted to move away due to a lot of complexities.

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 2]
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
 
 # Use Cases {#uc}
 
-## Media Streaming {#uc-streaming}
+## Streaming and Interactive Media {#uc-media}
```
Comment (sridharangirish91) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: IMHO, Even if the requirements are overlapping, Streaming and Interactive Media should be separate items, given they are 2 big use cases that comprise of most of the internet traffic. On the other hand, we can merge interactive media, accommodate delay and mixed traffic into one.

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 3]
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
...
-Use cases:
-
-  1. A mobile/roaming user prioritizes audio over video during a VoIP call to have a seamless meeting experience.
-
-## Accommodate-Delay Traffic {#uc-accommodate-delay}
+REQ-PACKET-PRIORITY: Packet priority relative to other packets in the transport flow (UDP 4-tuple).
+REQ-PACKET-DELAY: Metadata to indicate whether the packet can tolerate delay.
 
-Accommodate-Delay traffic includes content that is more resilient to buffering and delays (e.g., file copy, file download) compared to interactive traffic. This traffic can be buffered in favor of improved interactivity, without significant impact to the user experience. It is least sensitive to buffer bloating.
+## Mixed-Traffic and User Preferences {#uc-preferences}
```
Comment (sridharangirish91) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: User Preference might not belong here given that is agnostic of the traffic type but is an end user option. I am not sure if combining these 2 makes sense

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 4]
```diff
@@ -546,64 +472,15 @@ Taking into account these challenges, and despite the use case is described in t
...
-
-REQ-NRLP:
-: The network shall inform the host of the Rate limiting policies
-
-Use cases:
-
-  1. Performance Optimization: Some applications support some forms of bandwidth measurements (e.g., {{app-measurement}}) which feed how the content is accessed to using ABR. Complementing or replacing these measurements with explicit signals will improve overall network performance and can help optimize the data transfer. Signaling bandwidth availability allows hosts to avoid contributing to network congestion.
-
-  2. Dynamic Adaptation: When the network informs the host about available bandwidth, the host can dynamically adjust its data transmission rate.
+### Delay Tolerance between Flows (Inter-flow) {#interflow-delay}
```
Comment (sridharangirish91) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: Again, we should refrain from any inter-flow discussions IMO.

**Review by danwing — APPROVED** [review]

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 5]
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
 ~~~~~~~~
 {: #Figure-netshaper title=”Metadata and Network Shaping”}
 
-{{Figure-netshaper}} provides an outline of the interaction of application packets with network layer entities. Application layer signaling and feedback between client – server (A – in figure) adjusts rate over a period of several RTTs using feedback and congestion control algorithms.
+{{Figure-netshaper}} shows a bottleneck (access) Router on the path of packets from Server to Client.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
{{Figure-netshaper}} shows a bottleneck (access) router on the path of packets from Server to Client.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 6]
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
...
+A network shaper in the Router manages QoS of flows of multiple users and can buffer (delay), discard or apply other flow control rules. Application layer signaling and feedback between client – server (A – in figure) adjusts rate over a period of several RTTs using feedback and congestion control algorithms.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
A network shaper in the router manages QoS of flows of multiple users and can buffer (delay), discard, or apply other flow control rules. Application layer signaling and feedback between Client – Server (A – in the figure) adjust transmission rate over a period of several RTTs using feedback and congestion control algorithms.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 7]
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
...
 Congestion control algorithms are generally conservative and settle to a steady rate that avoids excessive packet loss.
 In networks where link conditions (between Client and Router) vary significantly at timescales well below the RTT, this results in unused (wasted) bandwidth at short timescales.
 There is some research {{5G-Octopus}} to indicate that media applications can obtain better QoE when sending at a higher rate (less conservative than current CCA) and the media application is willing to tolerate some packet loss or delay of low priority packets.
 Packet priority and tolerance to delay of packets in such a case would be provided on-path in a side channel associated to the downstream packet (B – in figure).
 The requirements for this server-to-network (S2N) metadata are described in {{server-network}}.
+Network shapers observe flows and apply policies to maximize performance but is not aware if there is a preference for one flow (UDP 4-tuple) over another flow belonging to the same user and network attachment (e.g., a subscriber connection, a 3GPP PDU session. See {{net-attach}} for more details).
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
Network shapers observe flows and apply policies to maximize performance but are not aware whether  there is a high preference for one flow (UDP 4-tuple) over another flow belonging to the same user and network attachment (e.g., a subscriber connection, a 3GPP PDU Session. See {{net-attach}} for more details).
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 8]
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
...
+The client can provide the (access) Router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in figure).
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
Clients may provide an (access) router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in Figure).
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 9]
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
...
+The client may also provide information to the (access) Router to drop lower priority marked packets of a flow (UDP 4-tuple) temporarily which can in turn redirect bandwidth to other flows of that network attachment.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
Clients may also provide information to an (access) router to drop, when resources are congested,  'lower priority'-marked packets of a flow (UDP 4-tuple) temporarily which can in turn allocate bandwidth to other flows of that same network attachment.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 10]
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
 
 # Use Cases {#uc}
 
-## Media Streaming {#uc-streaming}
+## Streaming and Interactive Media {#uc-media}
 
-Streaming video contains the occasional key frame ("I-frame") containing a full video frame.
+Streaming media packets carry audio or video between server and client.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
Streaming media packets carry audio or video between servers and clients.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 11]
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
...
-
-Accommodate-Delay traffic includes content that is more resilient to buffering and delays (e.g., file copy, file download) compared to interactive traffic. This traffic can be buffered in favor of improved interactivity, without significant impact to the user experience. It is least sensitive to buffer bloating.
-
-Requirement: REQ-PACKET-NATURE as defined previously.
-
-Use cases:
-
-  1. A remote desktop user prioritizes graphics update over an on-going file copy operation.
-  2. User application in foreground is given priority over downloading software updates.
+REQ-PACKET-PRIORITY: Packet priority relative to other packets in the transport flow (UDP 4-tuple).
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
REQ-PACKET-PRIORITY: Packet priority relative to other packets within the same transport flow (UDP 4-tuple).
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 12]
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
...
+REQ-PACKET-DELAY: Metadata to indicate whether the packet can tolerate delay.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
REQ-PACKET-DELAY: Metadata to indicate whether a packet can tolerate delay.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 13]
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
...
 
-## User Preferences {#uc-preferences}
+## Mixed-Traffic and User Preferences {#uc-preferences}
 
-A game or VoIP application may want to signal different metadata for the same type of packet in each direction.
-For example, for a game, video in the server-to-client direction might be more important than audio, whereas input devices (e.g., keystrokes) might be more important than audio.  Each user can have varied preferences for the same type of data originating from the server.
+Some content is more resilient to buffering and delays (e.g., file copy, file download) compared to interactive traffic. This traffic can be buffered in favor of improved interactivity, without significant impact to the user experience.
+For example, an on-going file copy operation flow (UDP 4-tuple) a remote desktop user can tolerate more delay than the flow carrying graphics updates.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
For example, an on-going file copy operation flow (UDP 4-tuple) of a remote desktop user can tolerate more delay than the flow carrying graphics updates.
```

**Review by boucadair — CHANGES_REQUESTED** [review]

**Review by danwing — APPROVED** [review]

**Review by boucadair — APPROVED** [review]

**Review by sridharangirish91 — APPROVED** [review]

**Comment by danwing** [comment]
Incorporated via #70.

**Pull request merged**